packages/dds/tree/src/feature-libraries/sequence-field/utils.ts

```diff
@@ -253,39 +290,82 @@ export function tryExtendMark(lhs: ObjectMark, rhs: Readonly<ObjectMark>): boole
     }
 
     if (
-        (type !== "MoveIn" && rhs.changes !== undefined) ||
-        (lhs.type !== "MoveIn" && lhs.changes !== undefined)
+        (type !== "MoveIn" && type !== "ReturnTo" && rhs.changes !== undefined) ||
+        (lhs as Modify | HasChanges).changes !== undefined
     ) {
         return false;
     }
 
     switch (type) {
-        case "Insert":
-        case "MoveIn": {
-            const lhsAttach = lhs as Insert | MoveIn;
-            if (isEqualPlace(lhsAttach, rhs)) {
-                if (rhs.type === "Insert") {
-                    const lhsInsert = lhsAttach as Insert;
-                    lhsInsert.content.push(...rhs.content);
+        case "Insert": {
+            const lhsInsert = lhs as Insert;
+            if (isEqualPlace(lhsInsert, rhs)) {
+                lhsInsert.content.push(...rhs.content);
+                return true;
+            }
+            break;
+        }
+        case "MoveIn":
+        case "ReturnTo": {
+            // TODO: Handle reattach fields
+            const lhsMoveIn = lhs as MoveIn | ReturnTo;
+            if (isEqualPlace(lhsMoveIn, rhs) && moveEffects !== undefined) {
+                const prevMerge = moveEffects.dstMergeable.get(lhsMoveIn.id);
+                if (prevMerge !== undefined) {
+                    moveEffects.dstMergeable.set(prevMerge, rhs.id);
```

Comment:
Is `moveEffects.dstMergeable.get(prevMerge)` expected to always be undefined here? If so, how do we know that is it?

Reply:
I'm also having trouble figuring out what it means for `prevMerge` to not be undefined. It seems to mean we've tried to merge `lhsMoveIn` with another mark already, but if we did, how come the lhs mark is not that other mark?

Reply:
This is intended to handle the case where we merge more than two adjacent changes. Suppose there are three mergeable moves, A, B, and C. We might first come across the move-outs, where we will record that A could be merged with B, and B with C on the source side, but we can't actually merge the move-outs yet because we don't know if the destination side is mergeable. When we process the move-ins we will merge B into A and then C into A. When we process the move-outs again in the next pass we will need to remember that A, B, and C all could be merged on the destination side. So while handling the destination merges, when we merge C into A we check that A was already merged with B, and mark that B and C are mergeable instead of A and C.